### PR TITLE
Run3-sim146Y Modify the code to accommodate trapezoid for HF volumes

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Modify the code to accommodate trapezoid for HF volumes

#### PR validation:

Tested by running SIM jobs with modified xml files

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
+1

File: Geometry/HcalCommonData/python/cmsExtendedGeometry2021XML_cfi.py [hunk 1]
```diff
@@ -234,8 +234,8 @@
         'Geometry/HcalCommonData/data/hcalouteralgo/v1/hcalouteralgo.xml',
         'Geometry/HcalCommonData/data/hcalforwardalgo.xml',
         'Geometry/HcalCommonData/data/average/hcalforwardmaterial.xml',
-        'Geometry/HcalCommonData/data/hcalSimNumbering/2021/v2/hcalSimNumbering.xml',
-        'Geometry/HcalCommonData/data/hcalRecNumbering/2021/v1/hcalRecNumbering.xml',
+        'Geometry/HcalCommonData/data/hcalSimNumbering/2021/v1/hcalSimNumbering.xml',
+        'Geometry/HcalCommonData/data/hcalRecNumbering/2021/hcalRecNumbering.xml',
```
Comment (perrotta) [inline Geometry/HcalCommonData/python/cmsExtendedGeometry2021XML_cfi.py:238]: The PR description does not explain why you are appling these fixes (?) to the code just merged with #42463

Comment (bsunanda) [reply Geometry/HcalCommonData/python/cmsExtendedGeometry2021XML_cfi.py:238]: I am keeping the old file for comparison. The fixed one should be in CMSCommonData

**Comment by bsunanda** [comment]
I responded to this in the PR page. Need this PR to be merged for further progress. Thanks
________________________________

**Comment by bsunanda** [comment]
@perrotta Please merge this PR

**Comment by rappoccio** [comment]
+1

**Comment by bsunanda** [comment]
When I used the current default files in Run3 and Phase2 scenarios I find we have too many hits in the HF. So while investigating that, I need to keep the old ones (which were there before changing to the current default) for a strict comparison. So I am keeping the old ones in Geometry/HCalCommonData and I keep on trying different combinations till I find (1) DDD and DD4hep give the same set of SD's; (2) the excess hits in the HF disappear.

**Pull request merged**